doc/man3/OSSL_HTTP_REQ_CTX.pod

```diff
@@ -33,18 +34,27 @@ OSSL_HTTP_REQ_CTX_set_max_response_length
  int OSSL_HTTP_REQ_CTX_add1_header(OSSL_HTTP_REQ_CTX *rctx,
                                    const char *name, const char *value);
 
+ int OSSL_HTTP_REQ_CTX_set_expected(OSSL_HTTP_REQ_CTX *rctx,
+                                    const char *content_type, int asn1,
+                                    int timeout, int keep_alive);
  int OSSL_HTTP_REQ_CTX_set1_req(OSSL_HTTP_REQ_CTX *rctx, const char *content_type,
-                                const ASN1_ITEM *it, ASN1_VALUE *req);
+                                const ASN1_ITEM *it, const ASN1_VALUE *req);
  int OSSL_HTTP_REQ_CTX_nbio(OSSL_HTTP_REQ_CTX *rctx);
- ASN1_VALUE *OSSL_HTTP_REQ_CTX_sendreq_d2i(OSSL_HTTP_REQ_CTX *rctx,
-                                           const ASN1_ITEM *it);
+ BIO *OSSL_HTTP_REQ_CTX_exchange(OSSL_HTTP_REQ_CTX *rctx);
 
  BIO *OSSL_HTTP_REQ_CTX_get0_mem_bio(const OSSL_HTTP_REQ_CTX *rctx);
+ size_t OSSL_HTTP_REQ_CTX_get_resp_len(const OSSL_HTTP_REQ_CTX *rctx);
  void OSSL_HTTP_REQ_CTX_set_max_response_length(OSSL_HTTP_REQ_CTX *rctx,
                                                 unsigned long len);
 
+ BIO *OSSL_HTTP_i2d(const ASN1_ITEM *it, const ASN1_VALUE *val);
+ ASN1_VALUE *OSSL_HTTP_d2i(BIO *res, const ASN1_ITEM *it);
+ int OSSL_HTTP_is_alive(const OSSL_HTTP_REQ_CTX *rctx);
+
 =head1 DESCRIPTION
 
+TODO(3.0) update
```

Comment:
Yes, that's part of #15053.

Reply:
If you like, I could add now to this PR also the updated doc in this file, and in `OSSL_HTTP_transfer.pod` etc.
This would mean 299 insertions, 174 deletions.
What do you prefer?

Reply:
I think it would be worth it as it is hard to review API without the actual doc of it. I had to look at the other PR for that.

Reply:
Now also the changes to the doc files are included.